docs/quality-assurance/bdd.md

```diff
@@ -0,0 +1,80 @@
+# Behavioural-Driven Design (BDD) at UKHO
+
+> "An approach that enhances communication between stakeholders and project team members by expressing product needs as concrete examples"
+
+This page contains information about BDD and how it can be used to aid how we develop software at UKHO.
+
+## What is BDD
+
+* A process to enable the entire team to better understand stories before they are played
+* BDD is about _collaboration_ - Product Owners (proxy for the customer), developers and testers communicating to gain an agreed understanding
+* Uses rules and examples to illustrate this understanding
+* When stories are played there is less ambiguity leading to better software development
+* BDD is NOT:
+  * Testing
+  * SpecFlow or Cucumber
+  * Just automated tests (tests are a downstream benefit)
+
+## Aims of BDD
+
+BDD aims to make the development of software more successful by:
+
+* Incorporating diverse perspectives when defining requirements
+* Creating a shared understanding in a delivery team of the required product changes
+* Reducing the amount of re-work required in the maintenance phase (by making the original change more complete)
+* Simplifying and shortening team Scrum events (refinement and planning)
+
+## Stages of BDD
+
+* Discovery - ensuring the team have a full understanding of a user requirement, using rules and examples to demonstrate this
+* Formulation - transforming the examples into documentation that describes how the product behaves (i.e Gherkin)
+* Automation - using the documentation to drive automated tests
+
+## Suggested BDD Approach
+
+* Discovery - Customer Representative brings requirements to the team as user stories (PBIs)
+* Discovery - The Example Mapping technique is applied to these stories by either the whole team or a “Three Amigos” subset
+* Discovery - Create rules and examples.. will drive out questions about the story, with the output a list of examples as acceptance criteria
```

Comment:
This sentence doesn't read well, maybe this would be better?
```suggestion
* Discovery - By creating rules and examples, you will drive out questions about the story, with the output a list of examples as acceptance criteria
```

Reply:
Re-worded